### PR TITLE
HAWQ-1612. Implement vectorized aggregate functions, such as sum/count/avg.

### PR DESCRIPTION
**Pull request by zhangjackey**

### CODE REVIEW

File: contrib/vexecutor/vagg.h [hunk 1]
```diff
@@ -0,0 +1,94 @@
+#ifndef VAGG_H
+#define VAGG_H
+
+#include "executor/execHHashagg.h"
+
+/* batch hashagg group linklist header */
+typedef struct GroupData {
+	HashAggEntry *entry;	// pointer to agg_hash_entry
+	int idx; 				// pointer to idx_list
```
Comment (interma) [inline contrib/vexecutor/vagg.h]: Refine my poor comments:
1. use /**/ instead //.
2. unify the ident.

File: contrib/vexecutor/vexecutor.c [hunk 2]
```diff
@@ -117,7 +122,7 @@ static void backportTupleDescriptor(PlanState* ps,TupleDesc td)
 		tuple = caql_getnext(pcqCtx);
 
 		if (!HeapTupleIsValid(tuple))
-			elog(ERROR, "cache lookup failed for type %u", oidtypeid);
+			continue;
```
Comment (interma) [inline contrib/vexecutor/vexecutor.c]: Why use `continue` instead of `elog ERROR` here?

File: contrib/vexecutor/vagg.h [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+} GroupData;
+
+/* batch hashagg group data */
+typedef struct BatchAggGroupData{
+	GroupData 	group_header[1024];	//group linklist header
```
Comment (interma) [inline contrib/vexecutor/vagg.h]: 1024, use a macro is better.

Comment (weinan003) [reply contrib/vexecutor/vagg.h]: variable **BATCHSIZE** in *vexecutor.c* indicate the number of slots the TupleBatch hold.

File: contrib/vexecutor/vexecutor.c (hunk 2 above)
Comment (zhangjackey) [reply contrib/vexecutor/vexecutor.c]: For the grouping aggregation, there may generate the tuple descriptor like this: {vtype,  vtype,  non-vtype, non-vtype}, we have to convert the vtype to non-vtype and skip the non-vtype which already exists.

**Pull request closed without merging**